src/main/java/io/github/connorhartley/guardian/util/StorageValue.java

```diff
@@ -24,181 +24,189 @@
-            ConfigurationNode node = storageDevice.getNode(this.key);
-            if(this.comment != null && node instanceof CommentedConfigurationNode) {
-                ((CommentedConfigurationNode)node).setComment(this.comment);
+    /* Internal Storage Values */
+
+    private <T> void setInternalValue(T storageHandler) {
+        if (storageHandler instanceof ConfigurationNode) {
+            ConfigurationNode node = ((ConfigurationNode) storageHandler).getNode(this.key.get());
+            if (this.comment != null && node instanceof CommentedConfigurationNode) {
+                ((CommentedConfigurationNode) node).setComment(this.comment);
             }
 
             if (this.modified) {
-                if (this.typeToken != null) {
+                if (this.valueTypeToken != null) {
                     try {
-                        node.setValue(this.typeToken, this.value);
+                        node.setValue(this.valueTypeToken, this.value);
                     } catch (ObjectMappingException e) {
                         e.printStackTrace();
-                        return false;
                     }
                 } else {
                     node.setValue(this.value);
                 }
                 this.modified = false;
             }
-
-            return true;
         }
-        return false;
     }
 
-    private <T extends ConfigurationNode, K extends String> Optional<V> getInternalValue(T storageDevice) {
-        if (storageDevice != null) {
-            ConfigurationNode node = storageDevice.getNode(this.key);
-            if(node.isVirtual()) {
+    private <T> E getInternalValue(T storageHandler) {
+        if (storageHandler instanceof ConfigurationNode) {
+            ConfigurationNode node = ((ConfigurationNode) storageHandler).getNode(this.key.get());
+
+            if (node.isVirtual()) {
                 this.modified = true;
             }
 
-            if(this.comment != null && node instanceof CommentedConfigurationNode) {
-                ((CommentedConfigurationNode)node).setComment(this.comment);
+            if (this.comment != null && node instanceof CommentedConfigurationNode) {
+                ((CommentedConfigurationNode) node).setComment(this.comment);
             }
 
             try {
-                if(typeToken != null)
-                    return Optional.of(node.getValue(this.typeToken, this.defaultValue));
-                else
-                    return Optional.of((V) node.getValue(new TypeToken(this.defaultValue.getClass()){}, this.defaultValue));
-            } catch(Exception e) {
-                return Optional.of(this.defaultValue);
+                if (this.valueTypeToken != null) {
+                    return node.getValue(this.valueTypeToken, this.defaultValue);
+                } else {
+                    return node.getValue(new TypeToken<E>(this.defaultValue.getClass()) {}, this.defaultValue);
+                }
+            } catch (Exception e) {
```

Comment:
Is this an acceptable case to occur? Should something be logged warning the owner this has happened?